back-nest/src/authentication/auth.controller.ts

```diff
@@ -10,26 +10,44 @@ import {
   UseGuards,
 } from '@nestjs/common';
 
-import { GetUser } from '../decorators/get-user.decorator';
+import { User } from '../decorators/user.decorator';
 import { AuthGuard } from './auth.guard';
 import { AuthService } from './auth.service';
+import { RegisterService } from './reg.service';
+import { SignUpDto } from '../dto/signup.dto';
 import { SignInDto } from '../dto/signIn.dto';
 import { UserDto } from '../dto/user.dto';
+import { CheckLoginDto } from '../dto/checkLogin.dto';
 
 @Controller('auth')
 export class AuthController {
-  constructor(private authService: AuthService) {}
+  constructor(
+    private readonly authService: AuthService,
+    private readonly regService: RegisterService,
+  ) {}
+
+  @Post('check-login')
+  async checkLogin(@Body() checkLogin: CheckLoginDto) {
+    const isAvalible = await this.regService.checkLoginAvailability(checkLogin);
+    return isAvalible;
+  }
```

Comment:
I don't think you need this endpoint. You can merge this logic into one /login endpoint